src/simplay-web/src/SimulationSpooler.ts

```diff
@@ -23,36 +26,88 @@ export class SimulationSpooler {
     createEntities(this.context);
   }
 
-  run(speedFactor = 1) {
-    throw Error('TODO implement');
+  private spoolTimestamp(timestamp: number) {
+    const events = this.simulationData.events.filter(
+      (event) => Math.round(event.timestamp) === timestamp
+    );
+    events.forEach((event) => {
+      event.execute(this.context);
+    });
   }
 
-  pause() {
-    throw Error('TODO implement');
+  async run() {
```

Comment:
As this is the external API, can we add the return type for these functions, or is it implicitly set to `Promise<void>`?